sdk/core/perf/src/base_test.cpp

```diff
@@ -138,72 +138,80 @@ namespace Azure { namespace Perf {
-              return h.first == "x-recording-id";
-            });
-        m_recordId = findHeader->second;
-        m_isPlayBackMode = true;
+        Azure::Core::_internal::ClientOptions clientOp;
+        clientOp.Retry.MaxRetries = 0;
+        ConfigureInsecureConnection(clientOp);
+        std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> policiesOp;
+        std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> policiesRe;
+        Azure::Core::Http::_internal::HttpPipeline pipeline(
+            clientOp, "PerfFw", "na", std::move(policiesRe), std::move(policiesOp));
+        Azure::Core::Context ctx;
+
+        //  Make one call to Run() before starting recording, to avoid capturing one-time setup
+        //  like authorization requests.
+        this->Run(ctx);
+
+        // Send start-record call
+        {
+          Azure::Core::Url startRecordReq(m_proxy);
+          startRecordReq.AppendPath("record");
+          startRecordReq.AppendPath("start");
+          Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Post, startRecordReq);
+          auto response = pipeline.Send(request, ctx);
+
+          auto const& headers = response->GetHeaders();
+          auto findHeader = std::find_if(
+              headers.begin(),
+              headers.end(),
+              [](std::pair<std::string const&, std::string const&> h) {
+                return h.first == "x-recording-id";
+              });
+          m_recordId = findHeader->second;
+        }
+
+        // Record one call to re-use response on all test runs
+        this->Run(ctx);
+
+        // Stop recording
+        {
+          Azure::Core::Url stopRecordReq(m_proxy);
+          stopRecordReq.AppendPath("record");
+          stopRecordReq.AppendPath("stop");
+          Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Post, stopRecordReq);
+          request.SetHeader("x-recording-id", m_recordId);
+          pipeline.Send(request, ctx);
+        }
+
+        // Start playback
+        {
+          Azure::Core::Url startPlayback(m_proxy);
+          startPlayback.AppendPath("playback");
+          startPlayback.AppendPath("start");
+          Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Post, startPlayback);
+          request.SetHeader("x-recording-id", m_recordId);
+          auto response = pipeline.Send(request, ctx);
+
+          auto const& headers = response->GetHeaders();
+          auto findHeader = std::find_if(
+              headers.begin(),
+              headers.end(),
+              [](std::pair<std::string const&, std::string const&> h) {
+                return h.first == "x-recording-id";
+              });
+          m_recordId = findHeader->second;
+          m_isPlayBackMode = true;
+        }
       }
     }
+    catch (std::exception const& e)
+    {
+      std::cout << "Exception thrown after setup: " << e.what() << std::endl;
```

Comment:
The general pattern across languages is perf tests should abort if there are any unexpected exceptions.  Is that what this code will do, or it will print the exception and continue running?

Reply:
This will print a message and rethrow. This needed to be handled separately from the main exception handler because it's executed in a separate thread from the main thread.

I've changed it to rethrow the exception, and added a debug only setting to disable the popup (MSVC specific).